compiler/frontend/src/string_to_rcst/whitespace.rs

```diff
@@ -231,16 +231,18 @@ mod test {
         assert_rich_ir_snapshot!(
             whitespaces_and_newlines("\n  foo", 0, true),
             @r###"
-        Remaining input: "  foo"
+        Remaining input: "foo"
         Parsed: Newline "\n"
+        Whitespace "  "
         "###
         );
         assert_rich_ir_snapshot!(
             whitespaces_and_newlines(" \n  foo", 0, true),
             @r###"
-        Remaining input: "  foo"
+        Remaining input: "foo"
         Parsed: Whitespace " "
         Newline "\n"
+        Whitespace "  "
         "###
         );
```

Comment:
Shouldn't the `Whitespace` here be wrapped in `CstError::TooMuchWhitespace` in the general case? (For parsed text, the output looks fine)